### PR TITLE
Blockchain applications list - Closes #1384

### LINKED ISSUE #1384: Create application list component for exploring blockchain applications
### Description

Create a component to explore the list of blockchain applications

### Acceptance Criteria

- Allow users to check the status, application name, avatar, pin status
- Allow users to search blockchain application based on name	
- Application list should have the pagination
- Allow user to pin/unpin by swipe right 
- Allow users to select rows to see more information on the selected application 
- Ensure [UI/UX](https://www.figma.com/file/j4xoLJZJUF7Wy6ZwSEagru/Mobile%3A-Applications-exploring-and-management?node-id=37%3A2853) design requirements are met
- Add required unit tests. 
- Add required e2e tests.

### Additional Information

- Fetch data from the given endpoint 
- Suggested component name `BlockchainApplicationList`
- We will implement api call in “Change API to v6” objective, mock the api response for now based on response format 
- Use the “usePinBlockchainApplication” hooks to set/delete pin by chainId
- Use the isPinned flag from the application
- On selecting an application user needs to navigate to the route with chainId params 

	
```
const {pins, setPin, deletePin} = usePinBlockchainApplication()
// Pins = [chainId]
```

### PR DESCRIPTION
**Pull request by ClementeSerrano**
### What was the problem?
This PR resolves #1384

### How was it solved?
- Add blockchain applications list component, navigation and connect it to tabs
- Add applications icon component and connect it with tabs navigator
- Enhance blockchain application mocks according to new backend schema
- Introduce get applications metadata query hook (`useGetApplicationsMetaQuery`) and connect it with `useBlockchainApplicationManagement`.
- Polish some redux blockchain applications store management. Add `setApplications` action for populating the applications state.
- WIP

### How was it tested?
Local environment, on iOS and Android Simulators.

### CODE REVIEW

File: src/hooks/useSearch.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+export function useSearch(props) {
+  const [term, setTerm] = useState('');
+  const [searching, setSearching] = useState(false);
+
+  const onSearchRef = useRef();
+  const onDebounceRef = useRef();
+
+  const debouncedTerm = useDebounce(term, (props && props.delay) || SEARCH_DEBOUNCE_DELAY);
+
+  onSearchRef.current = props && props.onSearch;
```
[the remark this answers is not in the document]
Comment (ClementeSerrano) [reply src/hooks/useSearch.js]: done!

File: src/modules/BlockchainApplication/components/BlockchainApplicationsList/index.js [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+import { deviceType } from 'utilities/device';
+import { colors, themes } from 'constants/styleGuide';
+import Input from 'components/shared/toolBox/input';
+import Icon from 'components/shared/toolBox/icon';
+import { IconButton } from 'components/shared/toolBox/button';
+import Swipeable from 'components/shared/Swipeable';
+import PinSvg from 'assets/svgs/PinSvg';
+import { usePinBlockchainApplication } from '../../hooks/usePinBlockchainApplication';
+import { useBlockchainApplicationManagement } from '../../hooks/useBlockchainApplicationManagement';
+import { useSearch } from '../../../../hooks/useSearch';
```
[the remark this answers is not in the document]
Comment (ClementeSerrano) [reply src/modules/BlockchainApplication/components/BlockchainApplicationsList/index.js]: done!

File: src/modules/BlockchainApplication/api/useGetApplicationsQuery.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  const [data, setData] = useState(undefined);
+  const [isLoading, setIsLoading] = useState(true);
+  const [isError, setIsError] = useState(undefined);
+
+  const dispatch = useDispatch();
+
+  function query() {
+    // TODO: Replace with real API call when backend is available.
+    return new Promise((resolve) => {
+      setTimeout(() => {
```
[the remark this answers is not in the document]
Comment (ClementeSerrano) [reply src/modules/BlockchainApplication/api/useGetApplicationsQuery.js]: done!

**Review by eniolam1000752 — CHANGES_REQUESTED** [review]

File: src/hooks/useDebounce.test.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
+import { renderHook, act } from '@testing-library/react-hooks';
+
```
[the remark this answers is not in the document]
Comment (ClementeSerrano) [reply src/hooks/useDebounce.test.js:2]: i use to separate by a line break the external library imports from the app imports

File: src/modules/BlockchainApplication/components/BlockchainApplicationsExplorer/index.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import BlockchainApplicationsList from '../BlockchainApplicationsList';
+
+export default function BlockchainApplicationsExplorer() {
+  return (
+    <>
+      <BlockchainApplicationsList />
+    </>
+  );
```
[the remark this answers is not in the document]
Comment (ClementeSerrano) [reply src/modules/BlockchainApplication/components/BlockchainApplicationsExplorer/index.js:9]: yeah, i left it in a fragment considering that i was to add more components there. Ill remove it now

**Review by eniolam1000752 — APPROVED** [review]

**Pull request merged**